### PR TITLE
Update Axon, Nx, and EXLA

### PR DESCRIPTION
**Pull request by seanmor5**
Requires: https://github.com/elixir-nx/axon/pull/477

### CODE REVIEW

File: lib/bumblebee/audio/speech_to_text.ex [hunk 1]
```diff
@@ -36,7 +36,7 @@ defmodule Bumblebee.Audio.SpeechToText do
           generate_fun.(params, inputs)
         end
       end,
-      batch_size: batch_size
+      [batch_size: batch_size] ++ defn_options
     )
```
Comment (josevalim) [inline lib/bumblebee/audio/speech_to_text.ex:40]:
```suggestion
      defn_options
    )
    |> Nx.Serving.process_options(batch_size: batch_size)
```

File: lib/bumblebee/text/generation.ex [hunk 2]
```diff
@@ -64,7 +64,7 @@ defmodule Bumblebee.Text.Generation do
           generate_fun.(params, inputs)
         end
       end,
-      batch_size: batch_size
+      [batch_size: batch_size] ++ defn_options
     )
```
Comment (josevalim) [inline lib/bumblebee/text/generation.ex:68]:
```suggestion
      defn_options
    )
    |> Nx.Serving.process_options(batch_size: batch_size)
```

File: lib/bumblebee/text/question_answering.ex [hunk 3]
```diff
@@ -49,7 +49,7 @@ defmodule Bumblebee.Text.QuestionAnswering do
           predict_fun.(params, inputs)
         end
       end,
-      batch_size: batch_size
+      [batch_size: batch_size] ++ defn_options
     )
```
Comment (josevalim) [inline lib/bumblebee/text/question_answering.ex:53]:
```suggestion
      defn_options
    )
    |> Nx.Serving.process_options(batch_size: batch_size)
```

File: lib/bumblebee/text/text_classification.ex [hunk 4]
```diff
@@ -45,7 +45,7 @@ defmodule Bumblebee.Text.TextClassification do
           scores_fun.(params, inputs)
         end
       end,
-      batch_size: batch_size
+      [batch_size: batch_size] ++ defn_options
     )
```
Comment (josevalim) [inline lib/bumblebee/text/text_classification.ex:49]:
```suggestion
      defn_options
    )
    |> Nx.Serving.process_options(batch_size: batch_size)
```

File: lib/bumblebee/text/token_classification.ex [hunk 5]
```diff
@@ -53,7 +53,7 @@ defmodule Bumblebee.Text.TokenClassification do
           scores_fun.(params, inputs)
         end
       end,
-      batch_size: batch_size
+      [batch_size: batch_size] ++ defn_options
     )
```
Comment (josevalim) [inline lib/bumblebee/text/token_classification.ex:57]:
```suggestion
      defn_options
    )
    |> Nx.Serving.process_options(batch_size: batch_size)
```

File: lib/bumblebee/text/zero_shot_classification.ex [hunk 6]
```diff
@@ -71,7 +71,7 @@ defmodule Bumblebee.Text.ZeroShotClassification do
           Utils.Nx.composite_unflatten_batch(scores, inputs.size)
         end
       end,
-      batch_size: batch_size
+      [batch_size: batch_size] ++ defn_options
     )
```
Comment (josevalim) [inline lib/bumblebee/text/zero_shot_classification.ex:75]:
```suggestion
      defn_options
    )
    |> Nx.Serving.process_options(batch_size: batch_size)
```

File: lib/bumblebee/vision/image_classification.ex [hunk 7]
```diff
@@ -48,7 +48,7 @@ defmodule Bumblebee.Vision.ImageClassification do
           scores_fun.(params, inputs)
         end
       end,
-      batch_size: batch_size
+      [batch_size: batch_size] ++ defn_options
     )
```
Comment (josevalim) [inline lib/bumblebee/vision/image_classification.ex:52]:
```suggestion
      defn_options
    )
    |> Nx.Serving.process_options(batch_size: batch_size)
```

File: mix.exs [hunk 8]
```diff
@@ -30,11 +30,11 @@ defmodule Bumblebee.MixProject do
...
     [
-      {:axon, "~> 0.3.1", axon_opts()},
+      {:axon, "~> 0.5", axon_opts()},
       {:tokenizers, "~> 0.2.0"},
-      {:nx, github: "elixir-nx/nx", sparse: "nx", override: true},
-      {:exla, github: "elixir-nx/nx", sparse: "exla", override: true, only: [:dev, :test]},
-      {:torchx, github: "elixir-nx/nx", sparse: "torchx", override: true, only: [:dev, :test]},
+      {:nx, "~> 0.5"},
+      {:exla, "~> 0.5", only: [:dev, :test]},
+      {:torchx, "~> 0.5", only: [:dev, :test]},
```
Comment (josevalim) [inline mix.exs:37]:
```suggestion
      {:nx, "~> 0.5.0"},
      {:exla, "~> 0.5.0", only: [:dev, :test]},
      {:torchx, "~> 0.5.0", only: [:dev, :test]},
```

File: mix.exs [hunk 9]
```diff
@@ -30,11 +30,11 @@ defmodule Bumblebee.MixProject do
 
   defp deps do
     [
-      {:axon, "~> 0.3.1", axon_opts()},
+      {:axon, "~> 0.5", axon_opts()},
```
Comment (josevalim) [inline mix.exs:33]:
```suggestion
      {:axon, "~> 0.5.0", axon_opts()},
```

File: mix.exs [hunk 10]
```diff
@@ -44,7 +44,7 @@ defmodule Bumblebee.MixProject do
       {:stb_image, "~> 0.6.0", only: :test},
       {:bypass, "~> 2.1", only: :test},
       {:ex_doc, "~> 0.28", only: :dev, runtime: false},
-      {:nx_signal, github: "polvalente/nx-signal", branch: "main"}
+      {:nx_signal, "~> 0.1"}
```
Comment (josevalim) [inline mix.exs:47]:
```suggestion
      {:nx_signal, "~> 0.1.0"}
```

File: lib/bumblebee/diffusion/vae_kl.ex [hunk 11]
```diff
@@ -453,7 +478,9 @@ defmodule Bumblebee.Diffusion.VaeKl do
         "decoder.post_quantization_conv" => "post_quant_conv",
         "decoder.input_conv" => "decoder.conv_in",
         "decoder.output_norm" => "decoder.conv_norm_out",
-        "decoder.output_conv" => "decoder.conv_out"
+        "decoder.output_conv" => "decoder.conv_out",
+        # not present
```
Comment (jonatanklosko) [inline lib/bumblebee/diffusion/vae_kl.ex:482]:
```suggestion
        # TODO: this is never loaded, add support for :ignored source name
```

Comment (jonatanklosko) [reply lib/bumblebee/diffusion/vae_kl.ex:482]: So the key parameter is serialized together with other model params?

Comment (seanmor5) [reply lib/bumblebee/diffusion/vae_kl.ex:482]: Yeah it is part of the model state in Axon, this is the same with dropout as well. I'm a bit surprised none of the other models complained the same way this one did. In the future I'd like to make a better distinction of state and params, but this is how it is for now

**Review by jonatanklosko — APPROVED** [review]

**Pull request merged**